### DISCUSSION TITLE
Hide parts of the URDF

### DISCUSSION THREAD

**Discussion by Woz4tetra** [foxglove/community#838] [Q&A]
Is there a way to hide parts of the URDF in foxglove? I have a model that has a lot of collision geometries that are not relevant for certain debugging activities and I'd like the ability to hide them. Is this possible to do? There doesn't seem to be a good way to "layer" URDFs. The structure seems to be one URDF model per robot.

**Comment by achim-k** [comment]
Hiding links of an URDF is not yet supported, but I think that shouldn't be too hard to add

Internal issue: https://linear.app/foxglove/issue/FG-5212/hide-urdf-links

---

> There doesn't seem to be a good way to "layer" URDFs. The structure seems to be one URDF model per robot.

What exactly do you mean by that?

**Reply by Woz4tetra** [reply]
I attempted to make certain parts of the URDF hidden by splitting the URDF into "layers". These layers are essentially different robots. But I found this doesn't play nice with ROS' joint states and static transform topics. The expectation seems to be one loaded URDF per robot.

**Comment by Woz4tetra** [comment]
I attempted to make certain parts of the URDF hidden by splitting the URDF into "layers". These layers are essentially different robots. But I found this doesn't play nice with ROS' joint states and static transform topics. The expectation seems to be one loaded URDF per robot.
